src/main.ts

```diff
@@ -226,22 +225,19 @@ async function injectViteIndexMiddleware(
 
     if (isIgnoredPath(req.path, req)) return next();
 
-    if (isStaticFilePath(req.path)) next();
-    else {
-      const indexPath = findClosestIndexToRoot(req.path, config.root);
-      if (indexPath === undefined) return next();
-
-      const template = fs.readFileSync(indexPath, "utf8");
-      let html = await server.transformIndexHtml(req.originalUrl, template);
-
-      try {
-        html = await getTransformedHTML(html, req);
-        res.send(html);
-      } catch (e) {
-        console.error(e);
-        res.status(500);
-        return next();
-      }
+    const indexPath = findFilePath(req.path, config.root);
+    if (indexPath === undefined) return next();
+
+    const template = fs.readFileSync(indexPath, "utf8");
```

Comment:
I feel like we need to find more meaningful name instead of `findFilePath` for that function. I'm not sure how those HTML files should be called, templates maybe?
So it would be called `findTemplateFilePath` or even `readTemplate` and it would replace those couple of lines

```suggestion
    const template = readTemplate(req.path, config.root);
    if (!template) return next();
```

What do you think about that?